src/main/java/org/damap/base/rest/storage/InternalStorageService.java

```diff
@@ -26,7 +48,169 @@ public List<InternalStorageDO> getAllByLanguage(String languageCode) {
             storageTranslation ->
                 internalStorageDOList.add(
                     InternalStorageDOMapper.mapEntityToDO(
-                        storageTranslation, new InternalStorageDO())));
+                        storageTranslation.getInternalStorageId(),
+                        new InternalStorageDO(),
+                        List.of(storageTranslation))));
     return internalStorageDOList;
   }
+
+  /**
+   * create a new internal storage option.
+   *
+   * @param data a {@link org.damap.base.rest.storage.InternalStorageDO} object
+   * @return a {@link org.damap.base.rest.storage.InternalStorageDO} object
+   */
+  @Override
+  @Transactional
+  public InternalStorageDO create(InternalStorageDO data) {
+    InternalStorageValidator.validateForCreation(data);
+
+    InternalStorage internalStorage =
+        InternalStorageDOMapper.mapDOToEntity(data, new InternalStorage());
+    internalStorage.persistAndFlush();
+
+    for (InternalStorageTranslationDO translationDO : data.getTranslations()) {
+      InternalStorageTranslation translation =
+          InternalStorageTranslationDOMapper.mapDOToTranslationEntityForCreation(
+              translationDO, internalStorage);
+      translation.setInternalStorageId(internalStorage);
+      translation.persistAndFlush();
+    }
+
+    return getInternalStorageById(internalStorage.id);
+  }
+
+  /**
+   * read a specific internal storage option.
+   *
+   * @param id a {@link java.lang.String} object
+   * @param queryParams a {@link jakarta.ws.rs.core.MultivaluedMap} object
+   * @return a {@link org.damap.base.rest.storage.InternalStorageDO} object
+   */
+  @Override
+  public InternalStorageDO read(String id, MultivaluedMap<String, String> queryParams) {
+
+    InternalStorage internalStorage = internalStorageRepo.findById(Long.parseLong(id));
+    List<InternalStorageTranslation> translations =
+        internalStorageTranslationRepo.getAllInternalStorageTranslationsByStorageId(
+            Long.parseLong(id));
+
+    if (internalStorage == null) {
+      throw new NotFoundException("No internal storage with ID " + id + " found");
+    }
```

Comment:
Does not work, also not with "new"

> reason: no instance(s) of type variable(s) X exist so that NotFoundException conforms to Supplier<? extends X>

Reply:
InternalStorage internalStorage = internalStorageRepo.findByIdOptional(Long.parseLong(id))
            .orElseThrow(() -> new NotFoundException("No internal storage with ID " + id + " found"));

    List<InternalStorageTranslation> translations =
            internalStorageTranslationRepo.getAllInternalStorageTranslationsByStorageId(
                    Long.parseLong(id));